config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-presubmits.yaml

```diff
@@ -209,3 +223,67 @@ presubmits:
       testgrid-tab-name: pr-e2e
       testgrid-alert-email: k8s-testing-clusterapi-vsphere+alerts@groups.vmware.com
       description: Runs e2e tests
+
+postsubmits:
+  kubernetes-sigs/cluster-api-provider-vsphere:
+
+  # Deploys images and binaries after all merges to master
+  - name: post-cluster-api-provider-vsphere-deploy
+    labels:
+      preset-dind-enabled: "true"
+      preset-cluster-api-provider-vsphere-gcs-creds: "true"
+    branches:
+    - ^master$
+    always_run: false
+    run_if_changed: '^(cmd|config|pkg|vendor)/|Dockerfile'
+    decorate: true
+    path_alias: sigs.k8s.io/cluster-api-provider-vsphere
+    max_concurrency: 1
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20190719-7cdf877-master
+        command:
+        - runner.sh
+        args:
+        - ./hack/release.sh
+        - -l
+        - -p
+        - -k
+        - /root/.capv/keyfile.json
+        # we need privileged mode in order to do docker in docker
+        securityContext:
+          privileged: true
+    annotations:
+      testgrid-dashboards: vmware-cluster-api-provider-vsphere, sig-cluster-lifecycle-cluster-api-provider-vsphere
+      testgrid-tab-name: post-deploy
+      testgrid-alert-email: k8s-testing-clusterapi-vsphere+alerts@groups.vmware.com
+      description: Pushes new images and binaries from master
+
+  # Deploys images and binaries for tagged releases
+  - name: post-cluster-api-provider-vsphere-release
+    labels:
+      preset-dind-enabled: "true"
+      preset-cluster-api-provider-vsphere-gcs-creds: "true"
+    branches:
+    - ^v(\d)+\.(\d)+\.(\d)+(-(alpha|beta|rc)\.(\d)+)?$
+    decorate: true
+    path_alias: sigs.k8s.io/cluster-api-provider-vsphere
+    max_concurrency: 1
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20190719-7cdf877-master
+        command:
+        - runner.sh
+        args:
+        - ./hack/release.sh
+        - -p
+        - -k
```

Comment:
Hi @codenrhoden,

Please consider removing the `-k /root/.capv/keyfile.json` and using an environment variable.

Reply:
Same comment as before.

My intended course of action is this:

1. Merge this PR, with the use of the file from the secret and `-k`
2. Open a couple of PRs against CAPV that remove all use of the old image locations
3. Once that is working, modify these jobs again to not export `GCR_KEY_FILE` from the old secret, but instead export it from the new one

However, nothing about the process from the `hack/release.sh` script will reference the old image location, and thus doesn't need the old keyfile. I was originally worried about needing to push images to the new location, *and* have to get e2e tests to pass. And that's still my concern. I want the option to do both until it all gets sorted out.

I'd still rather go this route first. it won't last long.